jffs2/libjffs2.c

```diff
@@ -609,21 +609,24 @@ static int libjffs2_create(void *info, oid_t *dir, const char *name, oid_t *oid,
 
 	switch (type) {
 		case otFile:
-			mode = S_IFREG | S_IRWXU | S_IRWXG | S_IRWXO;
+			mode |= S_IFREG;
 			ret = idir->i_op->create(idir, dentry, mode, 0);
 			break;
 		case otDir:
-			mode = S_IFDIR | S_IRWXU | S_IRWXG | S_IRWXO;
+			mode |= S_IFDIR;
 			ret = idir->i_op->mkdir(idir, dentry, mode);
 			break;
 		case otDev:
-			mode = S_IFCHR | S_IRWXU | S_IRWXG | S_IRWXO;
+			if (!(S_ISCHR(mode) || S_ISBLK(mode) || S_ISFIFO(mode))) {
+				mode &= ACCESSPERMS;
+				mode |= S_IFCHR;
+			}
```

Comment:
Access permissions are not forced for character device. Also could you comment on setting S_IFCHR bit for block  devices and FIFOs?

Reply:
The permission bits aren't forced for any new file. We take them from the `mode` parameter.
In case of a device file (`type == otDev`) if the file type bits in `mode` aren't correctly set we default to the character device.
So we're not really setting S_IFCHR for block or fifo device - we default to S_IFCHR if the file type isn't already set in the `mode` parameter.

Reply:
Misunderstanding- by `are not forced` I have meant that `mode` is not masked by `ACESSPERMS` in opposition to block and FIFO devices.

Reply:
Thanks for the clarification :) I'll add `mode` file type bits checks too for consistency.